### PR TITLE
FLEX-240 Add endpointRunner to support executing of custom endpoints …

### PR DESCRIPTION
**Pull request by mjsalinger**
…from Flex

### CODE REVIEW

File: test/lib/modules/endpointRunner.test.js [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+      endpoint.execute({}, (err, result) => {
+        should.not.exist(err);
+        result.should.containDeep([{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+        return done();
+      });
+    });
+
+    it('should include a body', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
```
Comment (samlevin) [inline test/lib/modules/endpointRunner.test.js]: Would be cleaner make this a constant, e.g. `BAAS_URL`

File: test/lib/modules/endpointRunner.test.js [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+        result.should.containDeep([{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+        return done();
+      });
+    });
+
+    it('should include empty object as body if no body passed', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .post(`/rpc/${environmentId}/custom/myEndpoint`, {})
```
Comment (samlevin) [inline test/lib/modules/endpointRunner.test.js]: No request body validation?

File: test/lib/modules/endpointRunner.test.js [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .matchHeader('authorization', authorization)
+        .post(`/rpc/${environmentId}/custom/myEndpoint`)
+        .reply(200, [{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+
+      const endpoint = this.runner({ useUserContext: true }).endpoint('myEndpoint');
+      endpoint.execute({}, (err, result) => {
+        should.not.exist(err);
+        result.should.containDeep([{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
```
Comment (samlevin) [inline test/lib/modules/endpointRunner.test.js]: Why not assign this test object to a variable (with dynamically-generated values) and use it both here and in the nock handler?

File: test/lib/modules/endpointRunner.test.js (hunk 2 above)
Comment (mjsalinger) [reply test/lib/modules/endpointRunner.test.js]: This test tests if the body is empty, not sure what you're asking...

**Pull request merged**